### PR TITLE
feat(contracts): dynamic v1 fee model

### PR DESCRIPTION
**Pull request by kevinhalliday**
Improve v1 fee model.

- make it handle changes in asset and gas prices
- charge base gas limit
- allow for protocol fee

task: https://app.asana.com/0/1206208509925075/1207171569957943

### CODE REVIEW

**Comment by kevinhalliday** [comment]
Couple things:

- did not update bindings, because changing e2e will be a bit of work. will leave for another PR
- I just changed FeeOracleV1, rather than introducing a FeeOracleV2. Because v1 is not live anywhere, I see no need to keep it around

File: contracts/src/protocol/FeeOracleV1.sol [hunk 1]
```diff
@@ -11,36 +11,133 @@ import { IFeeOracleV1 } from "../interfaces/IFeeOracleV1.sol";
...
+            _setToNativeRate(p.chainId, p.toNativeRate);
+        }
+    }
+
+    /**
+     * @notice Set the gas price for a destination chain.
+     */
+    function _setGasPrice(uint64 chainId, uint256 gasPrice) internal {
+        require(gasPrice > 0, "FeeOracleV1: no zero gas price");
+        require(chainId != 0, "FeeOracleV1: no zero chain id");
```
Comment (pavelnikolov) [inline contracts/src/protocol/FeeOracleV1.sol:111]: Can `chaindId` be negative?

**Review by pavelnikolov — APPROVED** [review]
LGTM

**Review by ttarsi — APPROVED** [review]

File: contracts/src/protocol/FeeOracleV1.sol (hunk 1 above)
Comment (ttarsi) [reply contracts/src/protocol/FeeOracleV1.sol:111]: nope, it's a uint

**Pull request merged**